crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -39,6 +39,34 @@ use tokio::{
 };
 use tracing::{error, trace};
 
+/// Limits for logs queries
+#[derive(Default, Debug, Clone, Copy)]
+pub struct QueryLimits {
+    /// Maximum number of blocks that could be scanned per filter
+    max_blocks_per_filter: Option<u64>,
+    /// Maximum number of logs that can be returned in a response
+    max_logs_per_response: Option<usize>,
+}
+
+/// RPC interface for ETH filter API, implementing only the `eth_getLogs` method
+#[async_trait::async_trait]
+pub trait EngineEthFilter {
+    /// Returns logs matching given filter object.
+    async fn logs(&self, filter: Filter) -> RpcResult<Vec<Log>>;
```

Comment:
this function should have the querylimits as params because this is what we need to expose